### PR TITLE
Type hinting autobidsportal

### PR DESCRIPTION
**Pull request by kaitj**
More of a cosmetic / surface-level PR here to type hint all of the main functionality as a way of familiarizing myself with the codebase. Should also be beneficial for future maintainers.

A lot of `pyright: ignore` fixes.

### CODE REVIEW

**Comment by kaitj** [comment]
@tkkuehn this should be good for a look. Some of the typehinting may not be completely correct - was mostly based off of some digging through the code.

File: autobidsportal/tasks.py [hunk 1]
```diff
@@ -106,15 +139,22 @@ def run_cfmm2tar_with_retries(
     ----------
     out_dir
         Directory to which to download tar files.
+
     study_instance_uid
         StudyInstanceUid to download
 
+    Returns
+    -------
+    tuple[Iterable[Iterable[str]], str]
```
Comment (tkkuehn) [inline autobidsportal/tasks.py:148]: Should make this concrete

**Review by tkkuehn — APPROVED** [review]
A couple of comments that are not necessarily actionable, this largely looks good

File: autobidsportal/tasks.py (hunk 1 above)
Comment (kaitj) [reply autobidsportal/tasks.py:148]: Oops, good catch. Typehint was updated but docstring wasn't.

File: autobidsportal/tasks.py [hunk 2]
```diff
@@ -106,15 +139,22 @@ def run_cfmm2tar_with_retries(
...
+        List of tar files associated with study id processed and log
+
     Raises
     ------
     Cfmm2tarTimeoutError
         If cfmm2tar times out too many times.
     """
     cfmm2tar_result, log = [], ""
+    # Attempt 5 times before timing out
```
[the remark this answers is not in the document]
Comment (kaitj) [reply autobidsportal/tasks.py:157]: Hmm, will keep that in mind. I think 5 is reasonable in anycase, but can always change this in the future if needed.

**Pull request merged**